opencti-platform/opencti-graphql/src/database/file-storage.js

```diff
@@ -0,0 +1,268 @@
+      Key: id
+    }));
+    return object.Body;
+  } catch (err) {
+    logApp.info('[OPENCTI] Cannot retrieve file from S3', { error: err });
+    return null;
+  }
+}
+
+function streamToString(stream) {
+  return new Promise((resolve, reject) => {
+    const chunks = [];
+    stream.on('data', (chunk) => chunks.push(chunk));
+    stream.on('error', reject);
+    stream.on('end', () => resolve(Buffer.concat(chunks).toString('utf8')));
+  });
+}
+
+export async function getFileContent(id) {
+  const object = await s3Client.send(new s3.GetObjectCommand({
+    Bucket: bucketName,
+    Key: id
+  }));
+
+  return streamToString(object.Body);
+}
+
+export function storeFileConverter(user, file) {
+  return {
+    id: file.id,
+    name: file.name,
+    version: file.metaData.version,
+    mime_type: file.metaData.mimetype,
+  };
+}
+
+export async function loadFile(user, filename) {
+  try {
+    const object = await s3Client.send(new s3.GetObjectCommand({
+      Bucket: bucketName,
+      Key: filename
+    }));
+
+    return {
+      id: filename,
+      name: decodeURIComponent(object.Metadata.filename || 'unknown'),
+      size: object.ContentLength,
+      information: '',
+      lastModified: object.LastModified,
+      lastModifiedSinceMin: sinceNowInMinutes(object.LastModified),
+      metaData: { ...object.Metadata, messages: [], errors: [] },
+      uploadStatus: 'complete'
+    };
+  } catch (err) {
+    if (err instanceof s3.NoSuchKey) {
+      throw DatabaseError('File not found', { user_id: user.id, filename });
+    }
+
+    throw err;
+  }
+}
+
+export function isFileObjectExcluded(id) {
+  const fileName = id.includes('/') ? R.last(id.split('/')) : id;
+  return excludedFiles.map((e) => e.toLowerCase()).includes(fileName.toLowerCase());
+}
+
+export async function rawFilesListing(user, directory, recursive = false) {
+  let truncated = true;
+  let pageMarker;
+  const objects = [];
+  const requestParams = {
+    Bucket: bucketName,
+    Prefix: directory || undefined,
+    Delimiter: recursive ? undefined : '/'
+  };
+
+  while (truncated) {
+    const response = await s3Client.send(new s3.ListObjectsV2Command(requestParams));
+
+    if (!response.Contents) {
+      truncated = false;
+      break;
+    }
+
+    objects.push(...response.Contents);
+    truncated = response.IsTruncated;
+    if (truncated) {
+      pageMarker = response.Contents.slice(-1)[0].Key;
+      requestParams.Marker = pageMarker;
+    }
+  }
+
+  return objects.reduce(async (items, item) => {
+    if (!isFileObjectExcluded(item.Key)) {
+      items.push(await loadFile(user, item.Key));
+    }
+    return items;
+  }, []);
+}
+
+export async function upload(user, path, fileUpload, metadata = {}) {
+  const { createReadStream, filename, mimetype, encoding = '', version = now() } = await fileUpload;
+  const readStream = createReadStream();
+
+  logAudit.info(user, UPLOAD_ACTION, { path, filename, metadata });
+
+  const key = joinPaths(path, filename);
```

Comment:
I start some testing and looks like the upload is no longer working with minio
The file is no more created inside a hierarchy of directory
To be replace with

> const key = \`${path}/${filename}`;